packages/react/src/components/Text/TextDirectionContext.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { createContext, MutableRefObject, useContext } from 'react';
+import { GetTextDirection, TextDir } from './TextDirection';
+
+export interface TextDirectionContextData {
+  direction?: TextDir;
+  getTextDirection?: MutableRefObject<GetTextDirection | undefined>;
+}
+
+export const TextDirectionContext = createContext<TextDirectionContextData>({});
+export const useTextDirectionContext = () =>
+  useContext<TextDirectionContextData>(TextDirectionContext);
```

Comment:
Yes this is correct, the refactor here was so that we can have all the types for the context in one place. The addition of the `useTextDirectionContext` was provide a convenience  method for the `useContext` so we didnt have to manually pass the typing context in everytime. The empty object instantiation here is from a pattern we use over in fusion-ui, since this context isnt really used too widely I was able to replace the null here with just an empty object. For other contexts that have wider usages, we can probably just incorporate `null` as a valid type for the contexts to prevent bigger refactors

Reply:
A type is required for the `useContext` otherwise when its used we get an object with an `any` type and have no idea what properties are valid for this context

Reply:
I definitely get that, and in a totally typescript product I'd 100% agree this is the correct approach, but that's unfortunately not the state of `carbon/react` ...yet!

My argument is that the lack of typing on that context is no worse than what is currently present in the JS codebase. The existing code functions as-is without types. I definitely believe that converting all the things is a great eventual goal, but I don't believe that now is necessarily the right time for that.

In a general sense, what the System squad has the capacity to take on at this point maintenance and new-development-wise is the introduction of type interfaces at the externally-exposed component level, but not anything beyond that at this time. That's why I'm erring more on the side of caution with respect to the scope of things we convert to typescript out of the gate.

Reply:
Im a little confused, are you suggesting that we not convert this file to TS at all or just remove the interface and change it to `any`? Im not sure how the addition of the interface is going to cause confusion in the JS code, I would actually think that it helps set the expectations on what we are able to access and set. Are we concerned that a developer might not necessarily know that they have to add a prop to the interface if we wanted to add something new to a context?

Reply:
Are you saying you'd rather
```
export interface TextDirectionContextData {
  direction?: TextDir;
  getTextDirection?: MutableRefObject<GetTextDirection | undefined>;
  [key: any]: any
}
```

So they can pass in additional unknown arguments?  

Is there an actual use case for that?  Personally I'd air on making it harder for people to change the underlying assumptions of what's in the context.  You shouldn't be overriding a "global" you don't own, that's generally a code smell, and carbon owns this context.